From c27e5acd01bc6686ec9bda642c9f8badb97c370e Mon Sep 17 00:00:00 2001
From: Xufeng Zhang <xufeng.zhang@windriver.com>
Date: Thu, 18 Aug 2011 17:59:56 +0800
Subject: [PATCH 06/12] p1022ds: DTS: Fix crypto fsl,multi-host-mode setting issue in AMP mode

In AMP mode, core1 first boot up, so core1 dts crypto node "fsl,multi-host-mode"
property should be "primary", and core0 should be "secondary"

Extracted from vendor drop QorIQ-NONDPAA-SDK-V1-20110429_ltib.iso
0104-p1022ds-DTS-Fix-crypto-fsl-multi-host-mode-setting-i.patch.

Signed-off-by: Jiang Yutang <b14898@freescale.com>
Integrated-by: Xufeng Zhang <xufeng.zhang@windriver.com>
---
 arch/powerpc/boot/dts/p1022ds_camp_core0.dts       |    2 +-
 arch/powerpc/boot/dts/p1022ds_camp_core0_36bit.dts |    2 +-
 arch/powerpc/boot/dts/p1022ds_camp_core1.dts       |    2 +-
 arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts |   16 +++++++++++++++-
 4 files changed, 18 insertions(+), 4 deletions(-)

diff --git a/arch/powerpc/boot/dts/p1022ds_camp_core0.dts b/arch/powerpc/boot/dts/p1022ds_camp_core0.dts
index 691f55b..164a614 100644
--- a/arch/powerpc/boot/dts/p1022ds_camp_core0.dts
+++ b/arch/powerpc/boot/dts/p1022ds_camp_core0.dts
@@ -237,7 +237,7 @@
 			fsl,channel-fifo-len = <24>;
 			fsl,exec-units-mask = <0x97c>;
 			fsl,descriptor-types-mask = <0x3a30abf>;
-			fsl,multi-host-mode = "primary";
+			fsl,multi-host-mode = "secondary";
 			fsl,channel-remap = <0x3>;
 		};
 
diff --git a/arch/powerpc/boot/dts/p1022ds_camp_core0_36bit.dts b/arch/powerpc/boot/dts/p1022ds_camp_core0_36bit.dts
index 79a8484..abe5070 100644
--- a/arch/powerpc/boot/dts/p1022ds_camp_core0_36bit.dts
+++ b/arch/powerpc/boot/dts/p1022ds_camp_core0_36bit.dts
@@ -297,7 +297,7 @@
 			fsl,channel-fifo-len = <24>;
 			fsl,exec-units-mask = <0x97c>;
 			fsl,descriptor-types-mask = <0x3a30abf>;
-			fsl,multi-host-mode = "primary";
+			fsl,multi-host-mode = "secondary";
 			fsl,channel-remap = <0x3>;
 		};
 
diff --git a/arch/powerpc/boot/dts/p1022ds_camp_core1.dts b/arch/powerpc/boot/dts/p1022ds_camp_core1.dts
index 30afea1..01c064e 100644
--- a/arch/powerpc/boot/dts/p1022ds_camp_core1.dts
+++ b/arch/powerpc/boot/dts/p1022ds_camp_core1.dts
@@ -312,7 +312,7 @@
 			fsl,channel-fifo-len = <24>;
 			fsl,exec-units-mask = <0x97c>;
 			fsl,descriptor-types-mask = <0x3a30abf>;
-			fsl,multi-host-mode = "secondary";
+			fsl,multi-host-mode = "primary";
 			fsl,channel-remap = <0x3>;
 		};
 
diff --git a/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts b/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts
index 0fdfed5..9862f24 100644
--- a/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts
+++ b/arch/powerpc/boot/dts/p1022ds_camp_core1_36bit.dts
@@ -260,7 +260,21 @@
 			};
 		};
 
-
+		crypto@30000 {
+			compatible = "fsl,sec3.3", "fsl,sec3.1","fsl,sec3.0",
+				     "fsl,sec2.4", "fsl,sec2.2","fsl,sec2.1",
+				     "fsl,sec2.0";
+			reg = <0x30000 0x10000>;
+			interrupts = <45 2 58 2>;
+			interrupt-parent = <&mpic>;
+			fsl,num-channels = <4>;
+			fsl,channel-fifo-len = <24>;
+			fsl,exec-units-mask = <0x97c>;
+			fsl,descriptor-types-mask = <0x3a30abf>;
+			fsl,multi-host-mode = "primary";
+			fsl,channel-remap = <0x3>;
+		};	
+	
 		sata@19000 {
 			compatible = "fsl,mpc8536-sata", "fsl,pq-sata";
 			reg = <0x19000 0x1000>;
-- 
1.7.0.4

